syntax = "proto3";

package microless.media.moviereview;
option go_package = "microless/media/proto/moviereview";

import "proto/data.proto";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

message UploadMovieReviewRequest {
    string movie_id = 1;
    string review_id = 2;
    google.protobuf.Timestamp timestamp = 3;
}

message ReadMovieReviewsRequest {
    string movie_id = 1;
    int32 start = 2;
    int32 stop = 3;
}

message ReadMovieReviewsRespond {
    repeated Review reviews = 1;
}

message CreateMovieRequest {
    string movie_id = 1;
}

service MovieReviewService {
    rpc UploadMovieReview (UploadMovieReviewRequest) returns (google.protobuf.Empty);
    rpc ReadMovieReviews (ReadMovieReviewsRequest) returns (ReadMovieReviewsRespond);
    rpc CreateMovie (CreateMovieRequest) returns (google.protobuf.Empty);
}